### PR TITLE
Add support for posting message replies

### PR DESCRIPTION
**Pull request by uzi**
Also converted to apiRequest based.

👓 @schwers @phil303

### CODE REVIEW

File: src/apis/MessagesEndpoint.es6.js [hunk 1]
```diff
@@ -23,44 +25,78 @@ const parseGetBody = (res, apiResponse, apiRequest) => {
...
 export default {
-  get(apiOptions, query) {
-    const { type } = query;
-    return runQuery(apiOptions, 'get', getPath(query), {}, query, parseGetBody);
+  get(apiOptions, data) {
+    const path = getPath(data);
+    const query = { ...data.query, raw_json: 1 };
+
+    return apiRequest(apiOptions, 'GET', path, { query })
+      .then(apiResponse => parseGetBody(apiResponse));
```
Comment (uzi) [inline src/apis/MessagesEndpoint.es6.js]: Just use `parseGetBody`

File: src/apis/MessagesEndpoint.es6.js [hunk 2]
```diff
@@ -23,44 +25,78 @@ const parseGetBody = (res, apiResponse, apiRequest) => {
...
-const parsePostBody = (res, apiResponse, apiRequest) => {
-  const { body } = res;
-  console.log(body);
+const parsePostBody = (apiResponse) => {
+  const { body } = apiResponse.response;
+  if (has(body, 'json.data.things.0.data')) {
+    const message = body.json.data.things[0];
+    const model = TYPE_MAP[message.kind].fromJSON(message.data);
+    apiResponse.addModel(model);
+    return model;
```
Comment (uzi) [inline src/apis/MessagesEndpoint.es6.js]: Done -- now it's one

File: src/apis/MessagesEndpoint.es6.js (hunk 1 above)
Comment (uzi) [inline src/apis/MessagesEndpoint.es6.js]: Done

**Comment by uzi** [comment]
💇

File: src/apis/MessagesEndpoint.es6.js [hunk 3]
```diff
@@ -23,44 +25,83 @@ const parseGetBody = (res, apiResponse, apiRequest) => {
...
 };
 
-const parsePostBody = (res, apiResponse, apiRequest) => {
-  const { body } = res;
-  console.log(body);
+const parsePostBody = (apiResponse) => {
+  const { body } = apiResponse.response;
+  if (has(body, 'json.data.errors') && body.json.errors.length) {
+    // There's a problem -- return the errors
+    return { errors: body.json.errors };
```
Comment (schwers) [inline src/apis/MessagesEndpoint.es6.js]: After the discussion in the editing post/comment patch, this should throw a `ValidationError`.

File: src/apis/MessagesEndpoint.es6.js [hunk 4]
```diff
@@ -1,16 +1,18 @@
...
 
 const TYPE_MAP = {
   't1': Comment,
   't3': Post,
   't4': Message,
 };
 
-const parseGetBody = (res, apiResponse, apiRequest) => {
-  const { body } = res;
+const parseGetBody = (apiResponse) => {
```
Comment (schwers) [inline src/apis/MessagesEndpoint.es6.js]: nit: don't need parens

File: src/apis/MessagesEndpoint.es6.js [hunk 5]
```diff
@@ -23,44 +25,83 @@ const parseGetBody = (res, apiResponse, apiRequest) => {
...
+  const { before, after } = body.data;
+  apiResponse.meta = { before, after };
+
+  return apiResponse;
 };
 
-const parsePostBody = (res, apiResponse, apiRequest) => {
-  const { body } = res;
-  console.log(body);
+const parsePostBody = (apiResponse) => {
```
Comment (schwers) [inline src/apis/MessagesEndpoint.es6.js]: nit: no parens

**Comment by schwers** [comment]
💅

File: src/apis/MessagesEndpoint.es6.js [hunk 6]
```diff
@@ -23,44 +25,83 @@ const parseGetBody = (res, apiResponse, apiRequest) => {
...
+  const { body } = apiResponse.response;
+  if (has(body, 'json.data.errors') && body.json.errors.length) {
+    // There's a problem -- return the errors
+    return { errors: body.json.errors };
+  } else if (has(body, 'json.data.things.0.data')) {
+    // We're replying to an existing thread -- the API gives us
+    // that message, so we return the model.
+    const message = body.json.data.things[0];
+    const model = TYPE_MAP[message.kind].fromJSON(message.data);
+    apiResponse.addModel(model);
```
Comment (uzi) [inline src/apis/MessagesEndpoint.es6.js]: Remnant of an earlier patch... will remove and test

**Comment by uzi** [comment]
💇

**Comment by schwers** [comment]
:fish:

**Pull request merged**